lib/razorpay/subscription.rb

```diff
@@ -0,0 +1,36 @@
+require 'razorpay/request'
+require 'razorpay/entity'
+
+module Razorpay
+  # Subscription API allows you to create and
+  # manage subscriptions with Razorpay
+  class Subscription < Entity
+    def self.request
+      Razorpay::Request.new('subscriptions')
+    end
+
+    def self.create(options)
+      request.create options
+    end
+
+    def self.fetch(id)
+      request.fetch id
+    end
+
+    def self.all(options = {})
+      request.all options
+    end
+
+    def self.cancel(id)
```

Comment:
Needs to accept optional data, documented [here](https://razorpay.com/docs/subscriptions/api/#cancelling-a-subscription)

Reply:
@amitmahbubani I am getting some errors testing the Api with `cancel_at_cycle_end` option, sharing on the support email with @StSarc and will fix as the api responds in expected way :+1: